WordPress/WordPress.xcodeproj/project.pbxproj

```diff
@@ -2032,6 +2032,11 @@
 		AE2F3129270B6DE200B2A9C2 /* NSMutableAttributedString+ApplyAttributesToQuotes.swift in Sources */ = {isa = PBXBuildFile; fileRef = AE2F3127270B6DE200B2A9C2 /* NSMutableAttributedString+ApplyAttributesToQuotes.swift */; };
 		AE3047AA270B66D300FE9266 /* Scanner+QuotedTextTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = AE3047A9270B66D300FE9266 /* Scanner+QuotedTextTests.swift */; };
 		AEE0828A2681C23C00DCF54B /* GutenbergRefactoredGalleryUploadProcessorTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = AEE082892681C23C00DCF54B /* GutenbergRefactoredGalleryUploadProcessorTests.swift */; };
+		B02D1EC72835348900F20359 /* pride-'22-icon-app-60x60@3x.png in Resources */ = {isa = PBXBuildFile; fileRef = B02D1EC22835348900F20359 /* pride-'22-icon-app-60x60@3x.png */; };
+		B02D1EC92835348900F20359 /* pride-'22-icon-app-60x60@2x.png in Resources */ = {isa = PBXBuildFile; fileRef = B02D1EC32835348900F20359 /* pride-'22-icon-app-60x60@2x.png */; };
+		B02D1ECB2835348900F20359 /* pride-'22-icon-app-83.5x83.5@2x.png in Resources */ = {isa = PBXBuildFile; fileRef = B02D1EC42835348900F20359 /* pride-'22-icon-app-83.5x83.5@2x.png */; };
+		B02D1ECD2835348900F20359 /* pride-'22-icon-app-76x76.png in Resources */ = {isa = PBXBuildFile; fileRef = B02D1EC52835348900F20359 /* pride-'22-icon-app-76x76.png */; };
+		B02D1ECF2835348900F20359 /* pride-'22-icon-app-76x76@2x.png in Resources */ = {isa = PBXBuildFile; fileRef = B02D1EC62835348900F20359 /* pride-'22-icon-app-76x76@2x.png */; };
```

Comment:
> pride-'22-icon-app-76x76@2x.png

👋 @frosty, the name of the files here includes a single quote `'` because we wanted to name the icon `Pride '22`. In my testing, the value of [`defaultIconName`](https://github.com/wordpress-mobile/WordPress-iOS/blob/e56bbded77a8cca0e39697a24008ea348f260120/WordPress/Classes/Utility/AppIcon.swift#L96) needed to coincide with the file name (according to [the logic here, which lowercases the display name and replaces spaces with hyphens](https://github.com/wordpress-mobile/WordPress-iOS/blob/08d12dfc094177f648f4330226cba9df56a1306c/WordPress/Classes/Utility/AppIcon.swift#L21)).
Do you know if there there a better way to do this that I'm missing? I'm a bit wary of the quote in the filename somehow causing problems, but so far it seems to be not causing any.

Reply:
I was a bit wary of the quote too, but as you said it seems to be working okay. Other options I can think of would be:

- Name it Pride 22 or Pride 2022
- Add another optional Info.plist value (like there is for 'requires border' and 'legacy icon') named something like 'Icon file name'. If this is provided, it can overwrite the logic you linked to above. This would allow specific icons to provide a different display name than their file name

Reply:
Thanks for confirming I wasn't missing an obvious solution here.

> Add another optional Info.plist value (like there is for 'requires border' and 'legacy icon') named something like 'Icon file name'. If this is provided, it can overwrite the logic you linked to above. This would allow specific icons to provide a different display name than their file name

This is my preferred option, but I think for now it's best to merge this as-is since it's already late in the beta. Doing the above would require a minor re-factor and testing all over again on iPhone and iPad.

Reply:
I've created an issue for this to follow-up on it at a later date, https://github.com/wordpress-mobile/WordPress-iOS/issues/18692.

Reply:
Marking this as resolved.